src/encoded/upgrade/dataset.py

```diff
@@ -731,3 +731,49 @@ def experiment_35_36(value, system):
+                else:
+                    value['notes'] = 'This elements_selection_method was previously ATAC-seq, but now renamed to be accessible genome regions.'
+            elif re.match('DNase-seq', element_selection_method):
+                new_elements_selection_method.append('DNase hypersensitive sites')
+                if 'notes' in value:
+                    value['notes'] = f'{value.get("notes")}. This elements_selection_method was previously DNase-seq, but now renamed to be DNase hypersensitive sites.'
+                else:
+                    value['notes'] = 'This elements_selection_method was previously DNase-seq, but now renamed to be DNase hypersensitive sites.'
+            elif re.match('GRO-cap', element_selection_method):
+                new_elements_selection_method.append('transcription start sites')
+                if 'notes' in value:
+                    value['notes'] = f'{value.get("notes")}. This elements_selection_method was previously GRO-cap, but now renamed to be transcription start sites.'
+                else:
+                    value['notes'] = 'This elements_selection_method was previously GRO-cap, but now renamed to be transcription start sites.'
+            elif re.match('point mutations', element_selection_method):
+                new_elements_selection_method.append('sequence variants')
+                if 'notes' in value:
+                    value['notes'] = f'{value.get("notes")}. This elements_selection_method was previously point mutations, but now renamed to be sequence variants.'
+                else:
+                    value['notes'] = 'This elements_selection_method was previously point mutations, but now renamed to be sequence variants.'
+            elif re.match('single nucleotide polymorphisms', element_selection_method):
+                new_elements_selection_method.append('sequence variants')
+                if 'notes' in value:
+                    value['notes'] = f'{value.get("notes")}. This elements_selection_method was previously single nucleotide polymorphisms, but now renamed to be sequence variants.'
+                else:
+                    value['notes'] = 'This elements_selection_method was previously single nucleotide polymorphisms, but now renamed to be sequence variants.'
+            elif re.match('transcription factors', element_selection_method):
+                new_elements_selection_method.append('TF binding sites')
+                if 'notes' in value:
+                    value['notes'] = f'{value.get("notes")}. This elements_selection_method was previously transcription factors, but now renamed to be TF binding sites.'
+                else:
+                    value['notes'] = 'This elements_selection_method was previously transcription factors, but now renamed to be TF binding sites.'
+    value['elements_selection_method'] = new_elements_selection_method
```

Comment:
Right now you are say that it should be upgraded even if the new elements selection method array is empty with no items. You should check for the length of the array.